Doc/library/argparse.rst

```diff
@@ -2002,7 +2002,15 @@ Exiting methods
 .. method:: ArgumentParser.exit(status=0, message=None)
 
    This method terminates the program, exiting with the specified *status*
-   and, if given, it prints a *message* before that.
+   and, if given, it prints a *message* before that. The user can override
+   this method to catching errors manually::
+
+    import sys
+    class ErrorCatchingArgumentParser(argparse.ArgumentParser):
+        def exit(self, status=0, message=None)
+            if status:
+                print('Catching an error!')
```

Comment:
I would use `message` here somewhere, as an example. I would also avoid the word "catch", as above:

```suggestion
                print('Exiting because of an error:', message)
```

Reply:
done